odoo/addons/base/models/res_company.py

```diff
@@ -346,6 +346,7 @@ def unlink(self):
         return res
 
     def write(self, values):
+        res = super().write(values)
```

Comment:
This will cause `companies_needs_l10n` to always be empty because the condition `company.country_id` cannot be true at the same time as `values.get('country_id')` anymore.

There might be issues with the cache invalidatation also, didn't double check